[RA2 Ch3] Scheduling pods with non-resilient apps (#2154)
cntt-n/CNTT: doc/ref_arch/kubernetes/chapters/chapter03.md

The architecture must support a way to isolate the compute resources of the
infrastructure itself from the workloads compute resources.

> <a name="3.2.1.8"></a>

Reviewer: Previous section is 3.2.1.5
```suggestion
<a name="3.2.1.6"></a>
```

> #### 3.2.1.8 Scheduling Pods with Non-resilient Applications

Reviewer: Previous section is 3.2.1.5
```suggestion
#### 3.2.1.6 Scheduling Pods with Non-resilient Applications
```

Non-resilient applications are sensitive to platform impairments on Compute like pausing CPU cycles (for example because of OS scheduler) or Networking like packet drops, reordering or latencies. Such applications need to be carefully scheduled on nodes and preferably still decoupled from infrastructure details of those nodes.

| # | Intensive on  | Not intensive on | Using hardware acceleration | Requirements for optimized pod scheduling |
|---|---|---|---|---|
| 1 | Compute | Networking (dataplane) | No | CPU Manager |

> | 2 | Compute | Networking (dataplane) | CPU instructions | #1 plus NFD |

Reviewer 2: ```suggestion
| 2 | Compute | Networking (dataplane) | CPU instructions | CPU Manager, NFD |
```

> | 3 | Compute | Networking (dataplane) | Fixed function acceleration, Firmware-programmable network adapters or SmartNICs | #1 plus Device Plugin |

Reviewer 2: ```suggestion
| 3 | Compute | Networking (dataplane) | Fixed function acceleration, Firmware-programmable network adapters or SmartNICs | CPU Manager, Device Plugin |
```

> | 4 | Networking (dataplane) | | No, or Fixed function acceleration, Firmware-programmable network adapters or SmartNICs  | Huge Pages (for DPDK-based applications), CPU Manager with configuration for isolcpus and SMT; Multiple interfaces; NUMA topology; Device Plugin |

Reviewer 2: ```suggestion
| 4 | Networking (dataplane) | | No, or Fixed function acceleration, Firmware-programmable network adapters or SmartNICs  | Huge Pages (for DPDK-based applications); CPU Manager with configuration for isolcpus and SMT; Multiple interfaces; NUMA topology; Device Plugin |
```

> | 5 | Networking (dataplane) | | CPU instructions | #4 plus NFD |

Reviewer 2: ```suggestion
| 5 | Networking (dataplane) | | CPU instructions | Huge Pages (for DPDK-based applications); CPU Manager with configuration for isolcpus and SMT; Multiple interfaces; NUMA topology; Device Plugin; NFD |
```
